cli/cmd/sync_v3.go

```diff
@@ -350,6 +346,17 @@ func syncConnectionV3(ctx context.Context, syncOptions syncV3Options) (syncErr e
 	// Read from the sync stream and write to all destinations.
 	isComplete := int64(0)
 
+	// Inactivity timeout: track last activity and add timeout goroutine
+	const inactivityTimeout = 1 * time.Minute
+	lastActivity := time.Now().Unix()
+
+	log.Info().Dur("timeout", inactivityTimeout).Msg("Inactivity timeout enabled - sync will stop if no activity detected")
+
+	// Helper function to update activity time
+	updateActivity := func() {
```

Comment:
We already have `remoteProgressReporter` (which is a `godebouncer`) sending progress info to the platform. `remoteProgressReporter.SendSignal()` is called on every `Sync_Response_Insert` message. Maybe we can extend that.